src/odemis/acq/align/test/spot_test.py

```diff
@@ -238,6 +241,59 @@ def stop_generate(self):
             pass
 
 
+class TestFindGridSpots(unittest.TestCase):
+    """
+    Unit test class to test the behavior of FindGridSpots in odemis.util.spot.
+    """
+
+    def test_find_grid_close_to_image_edge(self):
+        """
+        Create an image with a grid of 8 by 8 spots near the edge of the image. Then test if the spots are found
+        in the correct coordinates.
+        """
+        image = np.zeros((256, 256))
+        # set a grid of 8 by 8 points to 1
+        image[4:100:12, 4:100:12] = 1
+        spot_coordinates, translation, scaling, rotation = spot.FindGridSpots(image, (8, 8))
+        # create a grid that contains the coordinates of the spots
+        xv = np.arange(4, 100, 12)
+        xx, yy = np.meshgrid(xv, xv)
+        grid = np.column_stack((xx.ravel(), yy.ravel()))
+        np.testing.assert_array_almost_equal(np.sort(spot_coordinates, axis=1), np.sort(grid, axis=1), decimal=1)
+
+    def test_find_grid(self):
+        """
+        Create an image with a grid of 8 by 8 spots. Then test if the spots are found in the correct coordinates and
+        that the rotation, scaling and translation are correct.
+        """
+        image = np.zeros((256, 256))
+        # set a grid of 8 by 8 points to 1
+        image[54:150:12, 54:150:12] = 1
+        spot_coordinates, translation, scaling, rotation = spot.FindGridSpots(image, (8, 8))
+        self.assertAlmostEqual(rotation, 0, places=4)
+        # create a grid that contains the coordinates of the spots
+        xv = np.arange(54, 150, 12)
+        xx, yy = np.meshgrid(xv, xv)
+        grid = np.column_stack((xx.ravel(), yy.ravel()))
+        np.testing.assert_array_almost_equal(np.sort(spot_coordinates, axis=1), np.sort(grid, axis=1), decimal=2)
+        np.testing.assert_array_almost_equal(translation, np.array([96, 96]), decimal=3)
+        np.testing.assert_array_almost_equal(scaling, np.array([12, 12]), decimal=3)
+
+    def test_find_grid_on_image(self):
+        """
+        Load an image with known spot coordinates, test if the spots are found in the correct coordinates and
+        that the rotation, scaling and translation are correct.
+        """
+        grid_spots = np.load("grid_spots.npz")
```

Comment:
It'd be helpful if the filename of "grid_spots.npz" implied that it contains the results for "multiprobe01". It'd just suggest a long name like: "multiprobe01_grid_spots.npz".